src/Interpreters/Cache/FileCache.cpp

```diff
@@ -1349,6 +1354,49 @@ void FileCache::assertCacheCorrectness()
     });
 }
 
+void FileCache::applySettingsIfPossible(const FileCacheSettings & new_settings, FileCacheSettings & actual_settings)
+{
+    if (!is_initialized || shutdown)
+        return;
+
+    size_t add_download_threads = 0;
+    {
+        std::lock_guard lock(apply_settings_mutex);
+
+        if (new_settings == actual_settings)
+            return;
+
+        size_t background_download_queue_size_limit = metadata.getBackgroundDownloadQueueSizeLimit();
+        if (background_download_queue_size_limit != new_settings.background_download_queue_size_limit)
+        {
+            LOG_DEBUG(log, "Changing background_download_queue_size_limit from {} to {}",
+                      background_download_queue_size_limit, new_settings.background_download_queue_size_limit);
+
+            metadata.setBackgroundDownloadQueueSizeLimit(new_settings.background_download_queue_size_limit);
```

Comment:
Called for each element of `caches_by_name` – does this mean the last one wins?

Reply:
What do you mean? For each `caches_by_name` there are difference settings and `metadata` is different.